core/modules/search/search.admin.inc

```diff
@@ -193,14 +227,34 @@ function search_admin_settings_validate($form, &$form_state) {
  */
 function search_admin_settings_submit($form, &$form_state) {
   $config = config('search.settings');
-  // If these settings change, the index needs to be rebuilt.
-  if (($config->get('search_minimum_word_size') != $form_state['values']['search_minimum_word_size']) || ($config->get('search_overlap_cjk') != $form_state['values']['search_overlap_cjk'])) {
-    $config->set('search_minimum_word_size', $form_state['values']['search_minimum_word_size']);
-    $config->set('search_overlap_cjk', $form_state['values']['search_overlap_cjk']);
-    backdrop_set_message(t('The index will be rebuilt.'));
-    search_reindex();
+
+  // Perform node-related checks only if content search is enabled.
+  if (in_array('node', $form_state['values']['search_active_modules'], TRUE)) {
+    // If these settings change, the index needs to be rebuilt.
+    if ((isset($form_state['values']['search_minimum_word_size']) && $config->get('search_minimum_word_size') != $form_state['values']['search_minimum_word_size']) || (isset($form_state['values']['search_overlap_cjk']) && $config->get('search_overlap_cjk') != $form_state['values']['search_overlap_cjk']) || (null !== $config->get('content_types') && isset($form_state['values']['content_types']) && $config->get('content_types') !== $form_state['values']['content_types'])) {
+      $config->set('search_minimum_word_size', $form_state['values']['search_minimum_word_size']);
+      $config->set('search_overlap_cjk', $form_state['values']['search_overlap_cjk']);
+      backdrop_set_message(t('The index will be rebuilt.'));
+      search_reindex();
+    }
+    else {
+      $config->set('search_minimum_word_size', $config->get('search_minimum_word_size'));
+      $config->set('search_overlap_cjk', $config->get('search_overlap_cjk'));
```

Comment:
These `$config->set()`s are setting the config option to the value retrieved from the config option itself. Seems like an extraneous loop.